tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -379,49 +420,48 @@ ROOT::Experimental::Detail::RPageSourceDaos::RPageSourceDaos(std::string_view nt
    auto args = ParseDaosURI(uri);
    auto pool = std::make_shared<RDaosPool>(args.fPoolLabel);
    fDaosContainer = std::make_unique<RDaosContainer>(pool, args.fContainerLabel);
-}
 
+   auto [locator, descBuilder] = RDaosContainerNTupleLocator::LocateNTuple(*fDaosContainer, fNTupleName, *fDecompressor);
 
-ROOT::Experimental::Detail::RPageSourceDaos::~RPageSourceDaos() = default;
+   if (!locator.fAnchor.has_value() || !locator.fAnchor->fNBytesHeader)
+      throw ROOT::Experimental::RException(
+         R__FAIL("Requested ntuple '" + fNTupleName + "' is not present in DAOS container."));
 
+   fDescriptorBuilder = std::move(descBuilder);
+   fNTupleIndex = locator.GetIndex();
```

Comment:
Actually, it is preferable to defer this to `AttachImpl()`, as it is where most of the magic occurs (also in the file backend).  I think this change should also slightly simplify the code.